lib/SocketHelpers.cpp

```diff
@@ -102,6 +102,21 @@ static bool ReceiveBytes(const wil::unique_socket& socket, gsl::span<uint8_t> bu
 {
     while (buffer.size_bytes() > 0)
     {
+        // Wait until there is something to read (or fail after a timeout)
+        const timeval timeout{0, 500000}; // 0.5 sec timeout
+        fd_set read_set{};
+        FD_SET(socket.get(), &read_set);
+        const auto socket_ready = select(0, &read_set, nullptr, nullptr, &timeout);
+        if (socket_ready == SOCKET_ERROR)
+        {
+            THROW_LAST_ERROR_MSG("Error while waiting for a message");
```

Comment:
Do you want to get the specific error code for tracing purposes here?

Reply:
This already log the specific error code (from `WSAGetLastError`). WinSock use the pattern when on failure, a generic error code (`SOCKET_ERROR`) is returned, et the last error is set to the specific error code.